pkg/virt-handler/cmd-client/manager.go

```diff
@@ -0,0 +1,34 @@
+package cmdclient
+
+import v1 "kubevirt.io/api/core/v1"
+
+type ClientManager interface {
+	Create(vmi *v1.VirtualMachineInstance) (LauncherClient, string, error)
+	Socket(vmi *v1.VirtualMachineInstance) (string, error)
+	IsResponsive(socketPath string) bool
+}
```

Comment:
Since this interface is only used in a single place - please consider moving its definition closer to its consumer - `pkg/virt-handler/vm.go`.
There it could be unexported - `type clientManager interface`.

Reply:
While it is used only in one place, many places will benefit from it.

Special notes for your reviewer:
```
Note: This PR doesn't convert everything into a new abstraction and does not leverage it for tests. This will be follow-up work. The implementation is almost identical to the old one. We use the real implementation for the tests and that ensures we do not break tests and production code as well.
```